src/NuGet.Server.Core/Infrastructure/ServerPackageRepository.cs

```diff
@@ -128,31 +130,34 @@ public IQueryable<IServerPackage> GetPackages()
              * 2. If the store has no packages at all. This is so we pick up initial packages as quickly as
              *    possible.
              */
-            if (_needsRebuild || !_serverPackageStore.HasPackages())
+            if (_needsRebuild || !_serverPackageCache.IsEmpty())
```

Comment:
They didn't catch regression because I didn't modify the behavior. I'll fix this.